pkg/network/http/ebpf_gotls.go

```diff
@@ -109,7 +145,25 @@ func newGoTLSProgram(c *config.Config) *GoTLSProgram {
 		log.Errorf("System arch %q is not supported for goTLS", runtime.GOARCH)
 		return nil
 	}
-	return &GoTLSProgram{}
+	p := &GoTLSProgram{
+		procRoot: c.ProcRoot,
+		analysisCache: *ttlcache.New(
```

Comment:
can we limit the number of total entries?

Reply:
We can. That limit would be the maximum number of currently hooked GoTLS binaries. I am not sure what value would be a good default to use for that.

Reply:
tell me what is the memory consumption of each entry, and then we can decide on an appropriate value

Reply:
This is hard to guess precisely, as a part of it are the probe IDs, via the `hookedBinary` struct that serves as the value type in the cache, and there are some metadata for each.

Each "probe ID" is a struct containing 3 strings, with a generated UID, a function name and a section name.

From looking at the struct representing an item in the cache, and the cache itself, I have the following approximation:
- Metadata from the cache + inode number serving as the key: 98 bytes
- There should be 5 probes per entry. I approximated the size to be around 344 bytes, so that's a total of 1720 bytes.
- For each additional process we consume 4 more bytes to store its PID. 
- We should also add the size of the offsets data we put in the eBPF map, which is another 360 bytes per entry.
 
So that gives us:

`98 + 360 + 1720 + 4*number_of_processes` bytes
or `2178 + 4*number_of_processes` bytes.

So a binary hooked with only one process would take around 2.18KB. A binary with an 100 processes is 2.58KB. 
The vast majority of it is storing the probe IDs, but without those we can't ever detach the probes. The overhead of the cache excluding the entry key, is around 98 bytes per entry.